include/tuvx/linalg/linalg.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+namespace tuvx {
+namespace linalg {
+
+// tridiag matrix
+template <typename T> struct trid_mat {
+  int size;
+  std::vector<T> udiag; // upper diagonal
+  std::vector<T> ldiag; // lower diagonal
+  std::vector<T> mdiag; // main diagonal
+};
+
+template <typename T> void fill_rand_vec(std::vector<T> &x, int size);
```

Comment:
```suggestion
/// @brief Allocates a vector to a given size and fills it with random values
/// @param x Vector to allocate and fill
/// @param size Size of the vector
template <typename T> void FillRandomVector(std::vector<T> &x, int size);
```
We've been using Doxygen to document the code for MICM. @K20shores @sjsprecious - should we do the same for TUV-x C++ code?
(and similar for other functions here...)

Reply:
I like this idea and we probably should do it consistently for TUV-x as well. It is a good time to mention this since we just begin our work.

Reply:
@mattldawson @sjsprecious @K20shores I have added Doxygen style documentation to my code. Let me know if I did it correctly.

Reply:
i think you did it correctly and I appreciate those comments.